presto-raptorx/src/main/java/com/facebook/presto/raptorx/util/PrioritizedFifoExecutor.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx.util;
+
+import com.google.common.collect.ComparisonChain;
+import com.google.common.util.concurrent.ExecutionList;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.log.Logger;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is based on {@link io.airlift.concurrent.BoundedExecutor}.
+ */
+@ThreadSafe
+public class PrioritizedFifoExecutor<T extends Runnable>
```

Comment:
"ThreadPoolExecutor supports taking a priority queue. This util could be simplified."

I believe so, but I will not go ahead do it right now, since it's there for 6 years. And it only simplifies PrioritizedFifoExecutor, but not FifoRunnableTask. 

That "ThreadPoolExecutor" 's constructor was there at least since JAVA6, from 2006, when they did 6 years ago, should knew it already. Anyway, I will put this off for now